src/common/time/TimeUtils.cpp

```diff
@@ -55,11 +55,16 @@ constexpr int64_t kMaxTimestamp = std::numeric_limits<int64_t>::max() / 10000000
         return Status::Error("Invalid second number `%ld'.", kv.second.getInt());
       }
       dt.sec = kv.second.getInt();
+    } else if (kv.first == "millisecond") {
+      if (kv.second.getInt() < 0 || kv.second.getInt() > 999) {
+        return Status::Error("Invalid millisecond number `%ld'.", kv.second.getInt());
```

Comment:
Could you add more boundary value test cases? such as ms is 999 or 1000, us is 999 or 1000.

Reply:
Good catch. Done.